glfs.c

```diff
@@ -626,17 +627,58 @@ static int tcmu_glfs_write(struct tcmu_device *dev,
 static int tcmu_glfs_reconfig(struct tcmu_device *dev,
                               struct tcmulib_cfg_info *cfg)
 {
+	struct glfs_state *gfsp;
+	char *config;
+	int ret = -1;
+
 	switch (cfg->type) {
 	case TCMULIB_CFG_DEV_SIZE:
-		/*
-		 * Let glusterfs tools handle size checks
-		 */
-		return 0;
+
+		gfsp = calloc(1, sizeof(*gfsp));
+		if (!gfsp) {
+			goto notify;
+		}
+
+		config = tcmu_get_path(dev);
+		if (!config) {
+			tcmu_dev_err(dev, "tcmu_get_path() failed\n");
+			goto fail;
+		}
+
+		gfsp->fs = tcmu_create_glfs_object(dev, config, &gfsp->hosts);
+		if (!gfsp->fs) {
+			tcmu_dev_err(dev,
+			             "tcmu_create_glfs_object(config=%s) failed\n", config);
+			goto fail;
+		}
+
+		gfsp->gfd = glfs_open(gfsp->fs, gfsp->hosts->path, ALLOWED_BSOFLAGS);
```

Comment:
glfs_close() is missing. Will fix this part.

Rest I will test and update.

@lxbsz Thanks!

Reply:
Yes，right